test/functional/test_framework/mininode.py

```diff
@@ -1548,44 +1579,69 @@ def on_version(self, conn, message):
             conn.ver_recv = conn.ver_send
         conn.nServices = message.nServices
 
-    # Helper functions
-    ##################
+    # Connection helper methods
 
     def add_connection(self, conn):
         self.connection = conn
 
-    # Wrapper for the NodeConn's send_message function
+    def wait_for_disconnect(self, timeout=60):
+        test_function = lambda: not self.connected
+        assert wait_until(test_function, timeout=timeout)
+
+    # Message receiving helper methods
+
+    def sync(self, test_function, timeout=60):
```

Comment:
This function seems to duplicate a lot of what `wait_until` above does. Maybe it makes sense to do something like:

    def sync(self, test_function, timeout=60):
        if not wait_until(test_function, timeout=timeout):
            AssertionError("Sync failed to complete")